lib/msal-browser/src/utils/PopupUtils.ts

```diff
@@ -77,12 +103,39 @@ export class PopupUtils {
          * window.innerWidth displays browser window"s height and width excluding toolbars
          * using document.documentElement.clientWidth for IE8 and earlier
          */
-        const width = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
-        const height = window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;
-        const left = Math.max(0, ((width / 2) - (BrowserConstants.POPUP_WIDTH / 2)) + winLeft);
-        const top = Math.max(0, ((height / 2) - (BrowserConstants.POPUP_HEIGHT / 2)) + winTop);
+        const winWidth = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
+        const winHeight = window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;
+
+        if (popupWindowAttributes.popupPosition) {
+            if (popupWindowAttributes.popupPosition.top < 0 || popupWindowAttributes.popupPosition.top > winHeight) {
+                logger.error("Invalid popup window position. Popup window should not be positioned off-screen.");
+                popupWindowAttributes.popupPosition.top = 0;
```

Comment:
Reassigning input params. Can we move the `validHeight`, `validWidth` etc. vars to the top and overwrite with the defaults if they're not valid?

Reply:
Sure, done.